src/classes/Collection.cls

```diff
@@ -2,317 +2,494 @@ public with sharing class Collection {
 
 	private List<SObject> records;
 
+
+	// CONSTRUCT
```

Comment:
I'm of the view that comments should either add something useful or not be there.

Reply:
It was useful for structuring the source code for me when understanding. But its you code ;-)